README.md

```diff
@@ -1,2 +1,77 @@
 # artifact-upload-handler
+
 A secure webserver to handle uploading CI conda build artifacts to a specified conda channel.
+
+To use, simply clone this repo and run the Python webserver, observing the
+[Usage](#usage) and [Requirements](#requirements) instructions below.
+
+
+## Usage
+
+```
+usage: artifact_upload_handler.py [-h] -d WRITE_DIR -p PORT -e CONDA_EXE -c
+                                  CERTFILE -k KEYFILE -t TOKEN_HASH
+
+optional arguments:
+  -h, --help            show this help message and exit
+  -d WRITE_DIR, --write_dir WRITE_DIR
+                        directory to write artifacts to
+  -p PORT, --port PORT  webserver port number
+  -e CONDA_EXE, --conda_exe CONDA_EXE
+                        full path to conda executable
+  -c CERTFILE, --certfile CERTFILE
+                        full path to certificate file for SSL
+  -k KEYFILE, --keyfile KEYFILE
+                        full path to keyfile for SSL
+  -t TOKEN_HASH, --token_hash TOKEN_HASH
+                        secure token hash
+```
+
+
+### Usage Example 
+
+For example, to start a secure webserver running on port 9999 that will write
+to the (imaginary) conda channel at ``/path/to/conda/channel``:
+
+```
+$ python artifact_upload_handler.py -d /path/to/conda/channel \
+                                    -p 9999 \
+                                    -e /path/to/env/bin/conda \
+                                    -c /path/to/mycertfile.crt \
+                                    -k /path/to/mykeyfile.key \
+                                    -t eccd989b
```

Comment:
@dkillick Did you want to replace `eccd989b` with `mysecuretoken` (as per below) or vice verse, just to join up the lovely examples you've provided?

Reply:
Good idea!

Reply:
Actually - no, doing so would be wrong - see [this note further down](https://github.com/SciTools-incubator/artifact-upload-handler/pull/1/files#diff-04c6e90faac2675aa89e2176d2eec7d8R69).

To try and keep security as high as possible, the webserver takes the **hashed** token on startup, but the POST request to the server passes the unhashed secure token, i.e. from the CI. The unhashed token is salted and hashed by the webserver and these results are compared.